hub: add usage tab (#20774)
docker/docs: content/docker-hub/download-rate-limit.md

---
description: Learn about usage and rate limits for Docker Hub.
keywords: Docker Hub, pulls, download, limit, usage
title: Docker Hub usage and rate limits
---

Understanding your Docker Hub usage and rate limits helps you manage your and
your organization's usage effectively.

## View Docker Hub usage

You can download a CSV file of your or your organization's Docker Hub usage. To
download the file:

1. Sign in to [Docker Hub](https://hub.docker.com).

   If you want to download usage for all members of an organization, you must
   sign in to an account that is an owner for that organization. Otherwise,
   you can only view your own usage. 

> 2. In Docker Hub, select **Usage**.

Reviewer: Maybe link here too?

```suggestion
2. In Docker Hub, navigate to the [**Usage** page](https://hub.docker.com/usage).
```

3. In the drop-down, select whether to download your personal data or
   data for an organization.
4. In **From** and **To**, select a date range for the data.
5. Select **Send report to email** to have Docker email you a link to the data
   file. Note that email processing time may vary.

The file contains the following comma separated values.

| CSV column         | Definition                                                                                                                                                                                                                 | Usage guidance                                                                                                                                                                |
|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `datehour`           | The date and hour of the pull that resulted in the data transfer.                                                                                                                                                          | This helps in identifying peak usage times and patterns.                                                                                                                      |

Reviewer: ```suggestion
| `datehour`           | The date and hour (`yyyy/mm/dd/hh`) of the pull that resulted in the data transfer.                                                                                                                                                          | This helps in identifying peak usage times and patterns.                                                                                                                      |
```

| `user_name`          | The Docker ID of the user that pulled the image                                                                                                                                                                            | This lets organization owners track data consumption per user and manage resources effectively.                                                                               |
| `repository`         | The name of the repository of the image that was pulled.                                                                                                                                                                   | This lets you identify which repositories are most frequently accessed and consume most of the data transfer.                                                                 |

> | `pat_name`           | Name of the personal access token that was used for authentication with Docker CLI.                                                                                                                                        | Personal access tokens are usually used to authenticate automated tools (Docker Desktop, CI/CD tools, etc.). This is useful for identifying which automated system issued the pull. |

Reviewer: question: is `generated` == tokens generated by Docker Desktop? (should we call that out here in that case?)

| `digest`             | The unique image digest for the image.                                                                                                                                                                                     | This helps in identifying the image.                                                                                                                                          |
| `version_checks`     | The number of version checks accumulated for the date and hour of each image repository. Depending on the client, a pull request can do a version check to verify the existence of an image or tag without downloading it. | This helps identify the frequency of version checks, which you can use to analyze usage trends and potential unexpected behaviors.                                          |

> | `pulls`              | The number of pulls (HEAD & GET requests) accumulated for the date and hour of each image repository.                                                                                                                      | This helps identify the frequency of repository pulls, which you can use to analyze usage trends and potential unexpected behaviors.                                          |

Reviewer: is 1 pull == 1 `docker pull`? The "HEAD & GET requests" bit made me wonder if e.g. pulling an image with a lot of layers would result in many requests, and therefore a higher pull count than images with fewer layers? Maybe worth clarifying.

edit: I see that pull requests are defined in the next section on rate limits. Maybe it's good enough, or if we wanted to, we could probably have a section that defines more clearly what constitutes a pull, in terms of requests, that's separate from the rate limit section. Your call!

> | `pulls`              | The number of pulls (HEAD & GET requests) accumulated for the date and hour of each image repository.                                                                                                                      | This helps identify the frequency of repository pulls, which you can use to analyze usage trends and potential unexpected behaviors.                                                |

Reviewer 2: I think we should remove "(HEAD & GET requests)" so that this definition matches the definition of a pull that is provided further down the page.

docker-ratelimit-source: 192.0.2.1
```

In the previous example, the pull limit is 100 pulls per 21600 seconds (6
hours), and there are 76 pulls remaining.

#### I don't see any RateLimit headers

If you don't see any RateLimit header, it could be because the image or your IP

> is unlimited in partnership with a publisher, provider, or an open-source

Reviewer: we're pretty inconsistent with this but it should be "open source"

```suggestion
is unlimited in partnership with a publisher, provider, or an open source
```

web pages, APIs, and image pulls. The limit is applied per-IP, and while the
limit changes over time depending on load and other factors, it's in the order
of thousands of requests per minute. The overall rate limit applies to all users
equally regardless of account level.

You can differentiate between these limits by looking at the error code. The
"overall limit" returns a simple `429 Too Many Requests` response. The pull
limit returns a longer error message that includes a link to this page.

## How do I authenticate pull requests?

The following section contains information on how to sign in to Docker Hub to
authenticate pull requests.

### Docker Desktop

If you are using Docker Desktop, you can sign in to Docker Hub from the Docker
Desktop menu.

Select **Sign in / Create Docker ID** from the Docker Desktop menu and follow
the on-screen instructions to complete the sign-in process.

### Docker Engine

If you're using a standalone version of Docker Engine, run the `docker login`
command from a terminal to authenticate with Docker Hub. For information on how
to use the command, see [docker login](../reference/cli/docker/login.md).

### Docker Swarm

> If you're running Docker Swarm, you must use the `-- with-registry-auth` flag to

Reviewer: ```suggestion
If you're running Docker Swarm, you must use the `--with-registry-auth` flag to
```

authenticate with Docker Hub. For more information, see [Create a

> service](../reference/cli/docker/service/create.md/#create-a-service). If you

Reviewer: ```suggestion
service](../reference/cli/docker/service/create.md#with-registry-auth). If you
```
